pbm/log/log.go

```diff
@@ -2,77 +2,511 @@ package log
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
 
 	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/percona/percona-backup-mongodb/pbm/defs"
+	"github.com/percona/percona-backup-mongodb/pbm/errors"
 )
 
+// globalLogger is the only logger used in PBM Agent, PBM CLI and PBM SDK.
+var globalLogger = &loggerImpl{
```

Comment:
logger does not keep state. this is pre-initialized defaults to avoid `nil` values and checks for `nil` during logging. PBM Agents and CLI set proper handlers on init.